From fac1e5d5da70e7fb56c520eb2d9b6a6e3282459e Mon Sep 17 00:00:00 2001
From: Megidd <megiddgit@gmail.com>
Date: Sun, 14 Feb 2021 11:29:18 +0330
Subject: [PATCH 1/2] Fix CMake Error at CMakeLists.txt:125 (install)

When using such command:

cmake .. -DLIBNEST2D_HEADER_ONLY=OFF -DCMAKE_INSTALL_PREFIX=./installdir -DRP_ENABLE_DOWNLOADING=ON

This error is thrown:

CMake Error at CMakeLists.txt:125 (install):
  install TARGETS given no ARCHIVE DESTINATION for static library target
  "libnest2d_clipper_nlopt".

This commit fixes the error as far as tested.
---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c2f2bec..ab71296 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -124,6 +124,7 @@ export(EXPORT ${PROJECT_NAME}Targets
 
 install(TARGETS libnest2d libnest2d_headeronly ${LIBNAME} 
   EXPORT ${PROJECT_NAME}Targets
+  ARCHIVE DESTINATION lib
   RUNTIME DESTINATION bin 
 )
 

From 6be371cad6ecaaf0d11b541400c3c8d77e605cd7 Mon Sep 17 00:00:00 2001
From: Megidd <megiddgit@gmail.com>
Date: Sun, 14 Feb 2021 13:14:31 +0330
Subject: [PATCH 2/2] Also consider another error:

When running this command:

cmake .. -DLIBNEST2D_HEADER_ONLY=OFF -DCMAKE_INSTALL_PREFIX=./installdir -DRP_ENABLE_DOWNLOADING=ON -DBUILD_SHARED_LIBS=ON

This error is thrown:

CMake Error at CMakeLists.txt:125 (install):
  install TARGETS given no LIBRARY DESTINATION for shared library target
  "libnest2d_clipper_nlopt".

This commits resolves the error, as far as tested.
---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ab71296..1b9d4ed 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -125,6 +125,7 @@ export(EXPORT ${PROJECT_NAME}Targets
 install(TARGETS libnest2d libnest2d_headeronly ${LIBNAME} 
   EXPORT ${PROJECT_NAME}Targets
   ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION lib
   RUNTIME DESTINATION bin 
 )
 
